cub/device/dispatch/dispatch_histogram.cuh

```diff
@@ -251,6 +251,14 @@ public:
                     "The common type of `LevelT` and `SampleT` must be "
                     "trivially copyable.");
 
+      // An arithmetic type that's used for bin computation of integral SampleT, guaranteed to not
+      // overflow for (max_level - min_level) * scale.fraction.bins. Since we drop invalid samples
+      // of less than min_level, (sample - min_level) is guaranteed to be non-negative. We use the
+      // rule: 2^l * 2^r = 2^(l + r) to determine a sufficiently large type to hold the
+      // multiplication result
+      using IntArithmeticT = cub::detail::
+        conditional_t<(sizeof(SampleT) + sizeof(CommonT) <= sizeof(uint32_t)), uint32_t, uint64_t>;
```

Comment:
Optional: something like this might be easier to read:
```suggestion
      using IntArithmeticT = cub::detail::conditional_t<       //
        sizeof(SampleT) + sizeof(CommonT) <= sizeof(uint32_t), //
        uint32_t,                                              //
        uint64_t>;
```